src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1933,15 +1951,28 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
                 }
             }
 
-            tgtPrefUse = BuildUse(embOp2Node->Op(1));
-            srcCount += 1;
-
-            for (size_t argNum = 2; argNum <= numArgs; argNum++)
+            switch (intrinEmb.id)
             {
-                srcCount += BuildDelayFreeUses(embOp2Node->Op(argNum), embOp2Node->Op(1));
-            }
+                // Special-case, CreateBreakPropagateMask's op2 is the RMW node.
+                case NI_Sve_CreateBreakPropagateMask:
+                    assert(tgtPrefEmbOp2OfOp2);
+                    tgtPrefUse = BuildUse(embOp2Node->Op(2));
+                    srcCount += 1;
+                    srcCount += BuildDelayFreeUses(embOp2Node->Op(1), embOp2Node->Op(2));
+                    srcCount += BuildDelayFreeUses(intrin.op3, embOp2Node->Op(2));
```

Comment:
nit: `op3` should be contained here (as `BRKN` only supports `<Pg>/Z` and so `op3` can only be zero in the case we have `CreateBreakPropgateMask` as an embeded mask) and so `BuildDelayFreeUses` would be incorrect.

If `op3` is not `zero`, then we shouldn't have an embedded mask operand here.

Reply:
It's not "incorrect" because `BuildDelayFreeUses` will return `0` and not do anything if `intrin.op3` is contained.

But, since `op3` will always be contained and be zero, then we shouldn't need to use `BuildDelayFreeUses`. In fact, we shouldn't do anything to `intrin.op3` and instead just have asserts making sure that it is always contained and zero.

Reply:
I don't think that's a valid assumption even if that's the current behavior. This is a case where LSRA almost universally does `op->isContained() ? BuildUse() : BuildDelayFreeUse()` and I'm fairly certain that this was actually required for correctness in some scenarios (and that applies to both building the use and marking the node as having been consumed).